internal/handlers/lokistack_create_or_update.go

```diff
@@ -51,30 +54,51 @@ func CreateLokiStack(ctx context.Context, req ctrl.Request, k k8s.Client) error
 		return err
 	}
 	ll.Info("manifests built", "count", len(objects))
+
+	var errCount int32
 	for _, obj := range objects {
 		l := ll.WithValues("object_name", obj.GetName(),
 			"object_kind", obj.GetObjectKind(),
 			"object", obj)
 
 		obj.SetNamespace(req.Namespace)
-		setOwner(stack, obj)
-		if err := k.Create(ctx, obj); err != nil {
-			l.Error(err, "failed to create object")
-			// TODO requeue the event, but continue anyway
+
+		if err := ctrl.SetControllerReference(&stack, obj, s); err != nil {
+			l.Error(err, "failed to set controller owner reference to resource", "resource", obj.GetName())
+			errCount++
+			continue
+		}
+
+		var existing client.Object
+		key := client.ObjectKeyFromObject(obj)
+		if err := k.Get(ctx, key, existing); err != nil {
+			if !apierrors.IsNotFound(err) {
+				errCount++
```

Comment:
I think you should log this error like you do elsewhere so that it's clear what is happening.